### ISSUE TITLE
Query fails if it contains count distinct and avg (Presto 0.188)

### ISSUE THREAD

**Issue by ORBAT** [prestodb/presto#9336]
The following query fails with `Query 20171113_150356_31926_jw4ax failed: Unexpected response from http://172.18.36.178:18080/v1/task/20171113_150356_31926_jw4ax.2.7?summarize` with [this stack trace](https://gist.github.com/ORBAT/f71206e3662b4eddc22695e5614da58b)

**edit**: Session [looks like this](https://gist.github.com/ORBAT/a8ea0af881e730522437e9c6e05528a6)

```sql
with event_tbl (user_id, event_name, client_ts) as (values 
  ('user1', 'event1', 1510584396)
  , ('user1', 'event2', 1510584398)
  , ('user1', 'event3', 1510584400)
  , ('user1', 'event1', 1510594400)
  , ('user1', 'event2', 1510594401)
  , ('user1', 'event3', 1510594402)
  , ('user1', 'event4', 1510594403)
  , ('user2', 'event1', 1510584396)
  , ('user2', 'event2', 1510584398)
  , ('user2', 'event3', 1510584400)
  , ('user2', 'event1', 1510594400)
  , ('user2', 'event2', 1510594401)
  , ('user2', 'event3', 1510594402)
  , ('user2', 'event4', 1510594403)
)

, events as (
  select user_id, event_name, from_unixtime(client_ts) at time zone 'utc' as occurred_at
  from event_tbl
)

,bounds as (
  select user_id, occurred_at, (to_unixtime(occurred_at) - to_unixtime(lag(occurred_at,1) over (partition by user_id order by occurred_at)))/60 as time_to_last_event
  from events
)

, sessions as (
  select user_id
         , occurred_at as session_start
         , lead(occurred_at,1, timestamp '2020-01-01 00:00') over (partition by user_id
                                                                  order by occurred_at) as session_end
         , rank() over (partition by user_id
                       order by occurred_at) as session_number
  from bounds
  where time_to_last_event > 30
    or time_to_last_event is null
)

, user_events as (
  select events.user_id
         , events.occurred_at
         , sessions.session_number
         , events.event_name
         , row_number() over (partition by events.user_id, sessions.session_number
                            order by events.occurred_at) as event_number
   from
     sessions
   join events on events.user_id = sessions.user_id
   and events.occurred_at >= sessions.session_start
   and events.occurred_at < sessions.session_end
)

, user_session_lengths as (
  select date(occurred_at) as day, user_id, session_number, max(occurred_at) - min(occurred_at) as session_length
  from user_events
  group by 1, 2, 3
)

select day, count(distinct user_id) as num_users, avg(session_length) as avg_session_length from user_session_lengths group by 1;
```

However, if you replace the final select with any of the following, it works as expected:

- `select day, count(*) as total_sessions, count(distinct user_id) as num_users from user_session_lengths group by 1;`
- `select day, count(*) as total_sessions, avg(session_length) as avg_session_length from user_session_lengths group by 1;`
- `select day, avg(session_length) as avg_session_length from user_session_lengths group by 1;`
- `select day, count(distinct user_id) as num_users from user_session_lengths group by 1;`

**Comment by martint** [comment]
Does it happen if you set the optimize_mixed_distinct_aggregations session property to false?

`set session optimize_mixed_distinct_aggregations = false`

**Comment by martint** [comment]
Here's a simpler repro:

```sql
with t (a, b) as (
    values (1, interval '1' second)
) 
select count(distinct a), max(b) 
from t;
```

**Comment by martint** [comment]
This happens because OptimizeMixedDistinctAggregations produces a plan that contains an expression like `CAST(null AS interval day to second)`, which makes the SQL parser fail during plan deserialization in the worker nodes.

**Comment by martint** [comment]
Caused by https://github.com/prestodb/presto/issues/9338

**Comment by ORBAT** [comment]
As expected, setting `optimize_mixed_distinct_aggregations` to `false` fixes this. Thanks for the prompt reply!

**Comment by findepi** [comment]
@martint why is SQL parser needed to deserialize the plan on worker nodes? Perhaps there is a strong reason for this, but why don't we send AST?

**Comment by martint** [comment]
This should be fixed in the upcoming 0.190 version
